.github/workflows/pre-commit.yml

```diff
@@ -34,3 +34,6 @@ jobs:
           pull-repository: true
         env:
           NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: TypeScript compiler check
```

Comment:
Is this better being moved to `build.yaml` after the `Setup node` step? That way it's not duplicated in other workflows (like the unit test workflow, for example), and it would also be an extra check in the post-merge step?